### PR TITLE
pywal: init at 2.0.5

### PR DESCRIPTION
**Pull request by Fresheyeball**
###### Motivation for this change

I wanted nixos to be prettier

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/pywal/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> {} }:
+let
+  py = pkgs.python36Packages;
+in py.buildPythonPackage rec {
+  name = "pywal";
+  version = "2.0.4";
+  src = builtins.fetchTarball {
+    url = "https://github.com/dylanaraps/pywal/archive/${version}.tar.gz";
+  };
```
Comment (Mic92) [inline pkgs/misc/pywal/default.nix]: Use `fetchFromGitHub` here and add a checksum.

File: pkgs/misc/pywal/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  preCheck = ''
+    mkdir tmp
+    HOME=$PWD/tmp
+  '';
+  postCheck = ''
+    rm -r tmp
+  '';
+  propagatedBuildInputs = with pkgs; [ imagemagick feh ] ++ (with py; [ setuptools ]);
```
Comment (Mic92) [inline pkgs/misc/pywal/default.nix]: Please also add `meta` attributes.

Comment (Mic92) [reply pkgs/misc/pywal/default.nix]: if the program uses `imagemagick` or `feh` executables, adding those to `propagatedBuildInputs` will not work except inside a `nix-shell`. Use `makeWrapper` here instead to extend `PATH`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/sbt-extras/default.nix#L29

File: pkgs/misc/pywal/default.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> {} }:
+let
+  py = pkgs.python36Packages;
+in py.buildPythonPackage rec {
+  name = "pywal";
```
Comment (Mic92) [inline pkgs/misc/pywal/default.nix]: `name = "pyval-${version}";`

File: pkgs/misc/pywal/default.nix [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> {} }:
```
Comment (Mic92) [inline pkgs/misc/pywal/default.nix]: remove `pkgs` here and add `imagemagick`, `feh` and `python3Packages` explictly.

File: pkgs/misc/pywal/default.nix [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  src = builtins.fetchTarball {
+    url = "https://github.com/dylanaraps/pywal/archive/${version}.tar.gz";
+  };
+  preCheck = ''
+    mkdir tmp
+    HOME=$PWD/tmp
+  '';
+  postCheck = ''
+    rm -r tmp
+  '';
```
Comment (Mic92) [inline pkgs/misc/pywal/default.nix]: Cleanup can be skipped here. Nix take care of this.

**Review by Mic92 — CHANGES_REQUESTED** [review]

File: pkgs/misc/pywal/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/misc/pywal/default.nix]: `pname = "pwal";`

File: pkgs/misc/pywal/default.nix [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> {} }:
+let
+  py = pkgs.python36Packages;
+in py.buildPythonPackage rec {
```
Comment (dotlambda) [inline pkgs/misc/pywal/default.nix]: `buildPythonApplication`

File: pkgs/misc/pywal/default.nix [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  name = "pywal";
+  version = "2.0.4";
```
Comment (dotlambda) [inline pkgs/misc/pywal/default.nix]: Version 2.0.5 is now available.

File: pkgs/misc/pywal/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/misc/pywal/default.nix]: You could also use `fetchPypi`: https://pypi.org/project/pywal/.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/misc/pywal/default.nix [hunk 8]
```diff
@@ -1,18 +1,24 @@
...
+    sha256 = "0aqlbyk3nw4rd7dzp7vijla1r4skgpd7aar7nkqm2jdjmj3wq3x4";
   };
   preCheck = ''
     mkdir tmp
     HOME=$PWD/tmp
   '';
-  postCheck = ''
-    rm -r tmp
+  postFixup = ''
+    wrapProgram $out/bin/wal --prefix PATH : ${stdenv.lib.makeBinPath [ imagemagick feh ]}
```
Comment (dotlambda) [inline pkgs/misc/pywal/default.nix]: Please use `makeWrapperArgs` instead.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/misc/pywal/default.nix [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{ imagemagick, feh, python36Packages, fetchFromGitHub, makeWrapper, stdenv }:
+python36Packages.buildPythonApplication rec {
+  pname = "pywal";
+  version = "2.0.5";
+  meta = {
+    description = "Generate and change colorschemes on the fly. A 'wal' rewrite in Python 3.";
+    license = stdenv.lib.licenses.mit;
+    platforms = stdenv.lib.platforms.unix;
```
[the remark this answers is not in the document]
Comment (Fresheyeball) [reply pkgs/misc/pywal/default.nix]: I can. How would I state that here?

File: pkgs/misc/pywal/default.nix (hunk 8 above)
Comment (dotlambda) [reply pkgs/misc/pywal/default.nix]: https://github.com/NixOS/nixpkgs/search?q=makeWrapperArgs

File: pkgs/misc/pywal/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/misc/pywal/default.nix]: https://nixos.wiki/wiki/Nixpkgs#Becoming_a_Nixpkgs_maintainer

[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/misc/pywal/default.nix]: Don't forget to add `inherit (src.meta) homepage;`.

[the remark this answers is not in the document]
Comment (Fresheyeball) [reply pkgs/misc/pywal/default.nix]: @dotlambda done!

**Comment by Mr-SD** [comment]
Excuse my ignorance, but how do I use this? I've tried to install the package by adding it to my configuration file but it seems that it can't be found, I'm getting `error: undefined variable pywal`. I can see this branch has been merged into master, is this package not in the stable channel or is there a special workaround to install it?

**Pull request merged**